src/sinks/datadog/metrics/encoder.rs

```diff
@@ -757,6 +769,33 @@ mod tests {
         ));
     }
 
+    #[test]
+    fn encode_counter_with_interval_as_rate() {
+        // When a counter explicitly has an interval, we need to encode it as a rate. This means
+        // dividing the value by the interval (in seconds) and setting the metric type so that when
+        // it lands on the DD side, they can multiply the value by the interval (in seconds) and get
+        // back the correct total value for that time period.
+
+        let value = 423.1331;
+        let interval_ms = 10000;
+        let rate_counter = get_simple_rate_counter(value, interval_ms);
+        let expected_value = value / (interval_ms / 1000) as f64;
+        let expected_interval = interval_ms / 1000;
+
+        // Encode the metric and make sure we did the rate conversion correctly.
+        let result = generate_series_metrics(&rate_counter, &None, log_schema());
+        assert!(result.is_ok());
+
+        let metrics = result.unwrap();
+        assert_eq!(metrics.len(), 1);
+
+        let actual = &metrics[0];
+        assert_eq!(actual.r#type, DatadogMetricType::Rate);
```

Comment:
Would it be valuable add a test to ensure the `DatadogMetricType::Count` is set as the type, in the event that an interval is not defined?